_scripts/customJS/tokenParser.js

```diff
@@ -0,0 +1,575 @@
+                    for (let char of filterFormatString[0]) {
+                        if (filterChars.includes(char)) filter += char;
+                        if (formatChars.includes(char)) format += char;
+                    }
+                } else if (filterFormatString.length === 2) {
+                    // we have a;b
+                    // part a is parsed as a filter + format string; part b is parsed as chainFormat
+                    for (let char of filterFormatString[0]) {
+                        if (filterChars.includes(char)) filter += char;
+                        if (formatChars.includes(char)) format += char;
+                    } 
+                    for (let char of filterFormatString[1]) {
+                        if (formatChars.includes(char)) chainFormat += char;
+                    }
+                } else if (filterFormatString.length === 3) {
+                    // we have a;b;c
+                    // a is filter, b is the format, c is chainFormat
+                    for (let char of filterFormatString[0]) {
+                        if (filterChars.includes(char)) filter += char;
+                    } 
+                    for (let char of filterFormatString[1]) {
+                        if (formatChars.includes(char)) format += char;
+                    }
+                    for (let char of filterFormatString[2]) {
+                        if (formatChars.includes(char)) chainFormat += char;
+                    }
+                }
+
+                token.filter = remDup(filter);
+                token.format = remDup(format);
+            }
+
+        };
+
+        // clean up tokens //
+        // target, targetdate, and currentdate are all the same thing
+        if (token.token == "target" || token.token == "currentdate") token.token = "targetdate"
+        // length and age are the same //
+        if (token.token == "length") token.token = "age"
+        // end is shorthand for endStatus //
+        if (token.token == "end" || token.token == "endstatus") token.token = "endStatus"
+        // start is shorthand for startStatus //
+        if (token.token == "start" || token.token == "startstatus") token.token = "startStatus"
+        // subtype is the same as subtypeof //
+        if (token.token == "subtype") token.token = "subtypeof"
+        // typeof and type are the same as maintype //        
+        if (token.token == "maintype" || token.token == "type") token.token = "typeof"
+
+        if (token.format?.includes("U") && isFirst) {
+            token.format = token.format.replace("U", "u") 
+        } else { 
+            token.format = token.format?.replace("U", "")
+        }
+
+        if (token.format?.includes("A") && isFirst) {
+            token.format = token.format.replace("A", "a")
+        } else { 
+            token.format = token.format?.replace("A", "")
+        }
+
+        // all tokens are lower case //
+        token.token = token.token.toLowerCase()
```

Comment:
This makes sense but there are a few mixed cases above (ie endStatus) - just for clarity might be worth cleaning up.